src/vs/workbench/api/browser/mainThreadAuthentication.ts

```diff
@@ -536,4 +544,40 @@ export class MainThreadAuthentication extends Disposable implements MainThreadAu
 		}
 		return false;
 	}
+
+	async $promptForClientDetails(authorizationServerUrl: string): Promise<{ clientId: string; clientSecret?: string } | undefined> {
+		const clientId = await this.quickInputService.input({
+			title: nls.localize('clientIdPromptTitle', "OAuth Client ID Required"),
+			prompt: nls.localize('clientIdPrompt', "The authorization server '{0}' does not support automatic client registration. Please enter a client ID that has been registered for VS Code:", authorizationServerUrl),
```

Comment:
Since there will be a modal, you don't need to include the first sentence and also remove the "please".. keep it short